api/models/mainDao.js

```diff
@@ -0,0 +1,81 @@
+const appDataSource = require("./dataSource");
+
+const getUserCart = async (userId) => {
+  const result = await appDataSource.query(
+    `SELECT
+    c.id AS cartId,
+    u.id AS userId,
+    i.id AS itemId,
+    i.name AS itemName,
+    item_sizes.teabag_size AS itemSize,
+    c.quantity,
+    i.price * c.quantity AS totalPrice
+    FROM carts AS c
+    INNER JOIN users AS u ON u.id=c.user_id
+    INNER JOIN items AS i ON i.id=c.item_id
+    INNER JOIN item_sizes ON item_sizes.id=i.size_id
+    WHERE u.id=?`,
+    [userId]
+  );
+
+  return result;
+};
+
+const getUpdateCart = async (itemId, quantity, userId) => {
+  await appDataSource.query(
+    `UPDATE carts
+      SET quantity=?
+      WHERE item_id=? AND user_id=?
+    `,
+    [quantity, itemId, userId]
+  );
+
+  const updateResult = await appDataSource.query(
+    `SELECT
+    c.id AS cartId,
+    u.id AS userId,
+    i.id AS itemId,
+    i.name AS itemName,
+    item_sizes.teabag_size AS itemSize,
+    c.quantity,
+    i.price * c.quantity AS totalPrice
+    FROM carts AS c
+    INNER JOIN users AS u ON u.id=c.user_id
+    INNER JOIN items AS i ON i.id=c.item_id
+    INNER JOIN item_sizes ON item_sizes.id=i.size_id
+    WHERE u.id=?`,
+    [userId]
+  );
+  return updateResult;
+};
```

Comment:
리턴값으로 update가 적용된 데이터를 보내주다니 좋네요! 👍

다만 updateResult라는 변수는 위에서 작성된 getUserCart를 그대로 활용할 수 있을 것 같습니다.

delete도 마찬가지입니다.

```suggestion
const getUpdateCart = async (itemId, quantity, userId) => {
  await appDataSource.query(
    `UPDATE carts
      SET quantity=?
      WHERE item_id=? AND user_id=?
    `,
    [quantity, itemId, userId]
  );

  return getUserCart(userId)
};
```